docs/content/en/docs/yaml-crd-ref/taskdefinition.md

```diff
@@ -3,3 +3,310 @@ title: KeptnTaskDefinition
 description: Define tasks that can be run pre- or post-deployment
 weight: 89
 ---
+
+
+A `KeptnTaskDefinition` defines tasks
+that are run by the Keptn Lifecycle Toolkit
+as part of the pre- and post-deployment phases of a
+[KeptnApp](./app.md).
+
+## Yaml Synopsis
+
+```yaml
+apiVersion: lifecycle.keptn.sh/v?alpha?
+kind: KeptnTaskDefinition
+metadata:
+  name: <task-name>
+spec:
+  function:
+    inline | httpRef | functionRef:
+    [parameters:
+      map:
+        textMessage: "This is my configuration"]
+    [secureParameters:
+      secret: slack-token]
```

Comment:
```suggestion
    httpRef:
      url: "https://www.example.com/yourscript.js"
    parameters:
      key: "value"
    secureParameters:
      secret: slack-token
```

Reply:
I think the Synopsis should use pseudo-code.  People can copy/paste the Example code.